docs/references.md

```diff
@@ -0,0 +1,340 @@
+
+The `Reference` class provides properties to inspect and work with reference objects:
+
+```python
+from cicerone.references import Reference
+from cicerone.spec import Schema
+
+# Create a reference object
+ref = Reference(ref='#/components/schemas/Pet')
+
+# Check reference type
+print(f"Is local: {ref.is_local}")  # True
+print(f"Is external: {ref.is_external}")  # False
+
+# Get the JSON Pointer
+print(f"Pointer: {ref.pointer}")  # /components/schemas/Pet
+
+# Get pointer components
+print(f"Parts: {ref.pointer_parts}")  # ['components', 'schemas', 'Pet']
+
+# Resolve the reference to get the actual Schema object
+pet_schema = spec.resolve_reference(ref)
+assert isinstance(pet_schema, Schema)
+print(f"Pet schema type: {pet_schema.type}")
+```
+
+### OAS 3.1 Summary and Description
+
+In OpenAPI 3.1, Reference Objects can have `summary` and `description` fields:
+
+```python
+from cicerone.references import Reference
+
+ref = Reference(ref='#/components/schemas/User',
+    summary='User schema',
+    description='Represents a user in the system'
+)
+
+print(f"Reference: {ref.ref}")
+print(f"Summary: {ref.summary}")
+print(f"Description: {ref.description}")
+```
+
+## Advanced Reference Operations
+
+### Nested References
+
+Cicerone can automatically follow nested references:
+
+```yaml
+components:
+  schemas:
+    UserList:
+      type: array
+      items:
+        $ref: '#/components/schemas/User'
+    User:
+      $ref: '#/components/schemas/Person'
+    Person:
+      type: object
+      properties:
+        name:
+          type: string
+```
+
+```python
+# By default, follows nested references
+person = spec.resolve_reference('#/components/schemas/User')
+# Returns the Person schema (fully resolved)
+
+# Or stop at the first level
+user_ref = spec.resolve_reference('#/components/schemas/User', follow_nested=False)
+# Returns: {'$ref': '#/components/schemas/Person'}
+```
+
+### Circular Reference Detection
+
+Some schemas have circular references (e.g., tree structures):
+
+```yaml
+components:
+  schemas:
+    Node:
+      type: object
+      properties:
+        value:
+          type: string
+        children:
+          type: array
+          items:
+            $ref: '#/components/schemas/Node'  # Circular!
+```
+
+Cicerone detects and handles circular references:
+
+```python
+from cicerone.references import ReferenceResolver
+
+# Check if a reference is circular
+resolver = ReferenceResolver(spec)
+is_circular = resolver.is_circular_reference('#/components/schemas/Node')
+print(f"Is circular: {is_circular}")
+
+# When resolving, circular references are detected
+try:
+    # This will work for the Node schema itself
+    node = spec.resolve_reference('#/components/schemas/Node')
+    print("Node schema resolved successfully")
+    
+    # But attempting to fully resolve a circular chain will raise an error
+    # (Only when follow_nested would cause infinite recursion)
+except RecursionError as e:
```

Comment:
Do we want it to raise or just stop resolving to prevent infinite reference chains?